core/src/test/java/com/coderanch/blackjack/HandTest.java

```diff
@@ -0,0 +1,173 @@
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.QUEEN, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS)),
+                            21
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS)),
+                            0
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS)),
+                            0
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.NINE, Suit.CLUBS)),
+                            21
+                    )
+            );
+
+    /**
+     * Cards to test.
+     */
+    @DataPoints("objects")
+    public static final Set<Card> CARDS = Cards.getStandardDeck();
+
+
+    /**
+     * Tests that {@link Hand#withAdditionalCard(Card)} maintains the correct score.
+     *
+     * @param handTest
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_maintainsCorrectScore(HandTestArgument handTest) {
+        var hand = new Hand(handTest.getCards().get(0), handTest.getCards().get(1));
+        for (int i = 2; i < handTest.getCards().size(); i++) {
+            hand = hand.withAdditionalCard(handTest.getCards().get(i));
+        }
+        assertThat("The best score must match the target", hand.bestScore(), is(handTest.getTargetScore()));
+    }
+
+    private static final class HandTestArgument {
+
+        /**
+         * Cards included in the test.
+         */
+        private final List<Card> cards;
+
+        /**
+         * The target score the hand should have.
+         */
+        private final int targetScore;
+
+        HandTestArgument(List<Card> cards, int targetScore) {
+            this.cards = Collections.unmodifiableList(cards);
+            this.targetScore = targetScore;
+        }
+
+        List<Card> getCards() {
+            return cards;
+        }
+
+        int getTargetScore() {
+            return targetScore;
+        }
+    }
+
+    /**
+     * Tests that passing {@code null} for {@code card} when constructing a new hand causes an exception to be thrown.
```

Comment:
Misleading documentation. The assertion is not about `card`, but rather about the first card of the hand.